end_to_end_tests/golden-record/my_test_api_client/models/model_with_merged_properties.py

```diff
@@ -0,0 +1,111 @@
+import datetime
+from typing import Any, Dict, List, Type, TypeVar, Union
+
+from attrs import define as _attrs_define
+from attrs import field as _attrs_field
+from dateutil.parser import isoparse
+
+from ..models.model_with_merged_properties_string_to_enum import ModelWithMergedPropertiesStringToEnum
+from ..types import UNSET, Unset
+
+T = TypeVar("T", bound="ModelWithMergedProperties")
+
+
+@_attrs_define
+class ModelWithMergedProperties:
+    """
+    Attributes:
+        simple_string (Union[Unset, str]): extended simpleString description Default: 'new default'.
+        string_to_enum (Union[Unset, ModelWithMergedPropertiesStringToEnum]):
+        string_to_date (Union[Unset, datetime.date]):
+        number_to_int (Union[Unset, int]):
+        any_to_string (Union[Unset, str]):  Default: 'x'.
+    """
+
+    simple_string: Union[Unset, str] = "new default"
+    string_to_enum: Union[Unset, ModelWithMergedPropertiesStringToEnum] = UNSET
```

Comment:
@dbanty I'm not sure if there's a bug here or if I just don't understand the intended behavior, but-- I would've expected a default value to appear here. The schemas for this property were:

```
            "stringToEnum": {
              "type": "string",
              "default": "a"
            },
```

```
            "stringToEnum": {
              "type": "string",
              "enum": ["a", "b"]
            },
 ```

So, shouldn't it be `ModelWithMergedPropertiesStringToEnum.A` here instead of `UNSET`?

But... in `merge_properties_test.py`, I think `test_merge_string_with_string_enum` specifically checks that the string default is converted to an enum default in a scenario like this. So, is this being lost in the template logic? I'm still not super clear on the interactions between templates so I had trouble finding the relevant code.

Reply:
Well, I found the cause of that, but the fix requires a judgment call. The immediate problem was in this line in `merge_properties.py`:

```python
override_default = current.convert_value(override.default)
```

The reason that doesn't work is that `convert_value` expects a raw value, like a `str` or `int`, but `override.default` is a `Value` . When `EnumProperty.convert_value` sees this, it throws up its hands and just returns the same `Value`.

The underlying problem is that the `Value` class, which we're using to mean the Python representation of a value, doesn't actually remember the raw value it was constructed from. So if the default for a StringProperty is `'A'`, the Value is `'\'A\''` which won't work for an EnumProperty lookup. I think the cleanest way to fix that in the long run would be to change `Value` to be an object that remembers both its raw value and its Python representation. However... we're currently inconsistent in the use of `Value`, like there are places where `prop.default` is supposed to be a `Value` but it's actually the original int/bool/etc. (the type checker isn't catching this, I think because of the indirect way that we're calling class constructors). So fixing this would touch a lot of code (and might require some extra shenanigans to avoid breaking custom templates that expect this inconsistent typing).

A workaround to fix the immediate problem would be to add logic (either in `EnumProperty.convert_value`, or `StringProperty.default_as_raw`) for unquoting/unescaping a string. This is a kludge but it would keep the scope of the PR small.

Reply:
Related: I noticed that the unit tests are often setting the `default` attribute in a way that's inconsistent with the real runtime behavior. They're calling the factories in `conftest.py`, which all call the class constructors directly, skipping the `build` methods in the property classes (which are responsible for turning raw default values into `Value`s), so for instance `int_property_factory(default=3)` produces an IntProperty whose `default` is `3` rather than `Value("3")`. Those tests pass because their expectations are also using the wrong types. And the type checker isn't catching any of that because of the indirect way the factories instantiate the classes.

Reply:
OK, I've pushed a workaround for this.